rpc: split URL generation code to a separate lightweight package (cli refactor sequence 2/n)

knz opened the pull request:
PR split from https://github.com/cockroachdb/cockroach/pull/82020 to ease review.

knz committed:
security,rpc: avoid depending on CertManager in LoadSecurityOptions

This is needed to move the LoadSecurityOptions code to a lightweight
package independent of 'security'.

Release note: None

knz committed:
rpc: finish connecting LoadSecurityOptions to securityassets

Release note: None

knz committed:
rpc: detach LoadSecurityOptions and PGURL from SecurityContext

This paves the way to moving them to a standalone package.

Release note: None

knz committed:
cli,rpc: detach LoadSecurityOptions and PGURL from `base`

This paves the way further to move them to a separate package.

Release note: None

knz committed:
rpc: move LoadSecurityOptions and PGURL to a new lightweight package

This allows clarifies their scope as separate from the RPC logic.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82069)
<!-- Reviewable:end -->

knz committed:
clientsecopts: rename PGURL to MakeURLForServer

We have enough things called "PGURL" as it is. This rename clarifies
the purpose.

Release note: None

knz commented:
(NB: Holding off on merging this despite the review approval, since we can't merge intermediate PRs in a sequence without dropping the entire sequence off github.)
